pkg/pool-hooks/test/foundry/DirectionalFeeHookExample.t.sol

```diff
@@ -0,0 +1,319 @@
+        // Since Protocol Swap Fee is 0 (was not set in this test) all swap fee amount returned to the pool
+        assertEq(
+            balancesBefore.poolTokens[usdcIdx] - balancesAfter.poolTokens[usdcIdx],
+            expectedAmountOut - swapFeeAmount,
+            "Pool USDC balance is wrong"
+        );
+
+        // Vault Balances (Must reflect pool)
+        assertEq(
+            balancesAfter.vaultTokens[daiIdx] - balancesBefore.vaultTokens[daiIdx],
+            daiExactAmountIn,
+            "Vault DAI balance is wrong"
+        );
+        assertEq(
+            balancesBefore.vaultTokens[usdcIdx] - balancesAfter.vaultTokens[usdcIdx],
+            expectedAmountOut - swapFeeAmount,
+            "Vault USDC balance is wrong"
+        );
+
+        // Vault Reserves (Must reflect vault balances)
+        assertEq(
+            balancesAfter.vaultReserves[daiIdx] - balancesBefore.vaultReserves[daiIdx],
+            daiExactAmountIn,
+            "Vault DAI reserve is wrong"
+        );
+        assertEq(
+            balancesBefore.vaultReserves[usdcIdx] - balancesAfter.vaultReserves[usdcIdx],
+            expectedAmountOut - swapFeeAmount,
+            "Vault USDC reserve is wrong"
+        );
+    }
+
+    // Test fee when pool is getting unbalanced
+    function testSwapUnbalancingPoolFee() public {
+        // Swap to meaningfully take the pool out of equilibrium
+        uint256 daiExactAmountIn = poolInitAmount / 2;
+
+        BaseVaultTest.Balances memory balancesBefore = getBalances(address(lp));
+        // Since there's no rate providers, and all tokens are 18 decimals, scaled18 and raw values are equal.
+        uint256[] memory balancesScaled18 = balancesBefore.poolTokens;
+
+        // Calculate the expected amount out (amount out without fees).
+        uint256 poolInvariant = StableMath.computeInvariant(
+            DEFAULT_AMP_FACTOR * StableMath.AMP_PRECISION,
+            balancesScaled18
+        );
+        uint256 expectedAmountOut = StableMath.computeOutGivenExactIn(
+            DEFAULT_AMP_FACTOR * StableMath.AMP_PRECISION,
+            balancesScaled18,
+            daiIdx,
+            usdcIdx,
+            daiExactAmountIn,
+            poolInvariant
+        );
+
+        // Call dynamic fee hook to fetch the expected swap fee percentage
+        (, uint256 expectedSwapFeePercentage) = DirectionalFeeHookExample(poolHooksContract).onComputeDynamicSwapFee(
+            IBasePool.PoolSwapParams({
+                kind: SwapKind.EXACT_IN,
+                amountGivenScaled18: daiExactAmountIn,
+                balancesScaled18: balancesScaled18,
+                indexIn: daiIdx,
+                indexOut: usdcIdx,
+                router: address(0), // The router is not used by the hook
+                userData: bytes("") // User data is not used by the hook
+            }),
+            pool,
+            SWAP_FEE_PERCENTAGE
+        );
+
+        // Swap DAI for USDC, bringing pool nearer balance
```

Comment:
Isn't this comment backwards?

Reply:
Done! It was not nearer balance, but nearer equilibrium. Still, I refactored the comment to make it clearer.